src/main/java/io/codelex/flightplanner/common/inDB/inDBService.java

```diff
@@ -0,0 +1,102 @@
+package io.codelex.flightplanner.common.inDB;
+
+import io.codelex.flightplanner.common.FlightServiceInterface;
+import io.codelex.flightplanner.domain.Airport;
+import io.codelex.flightplanner.domain.Flight;
+import io.codelex.flightplanner.dto.AddFlightRequest;
+import io.codelex.flightplanner.dto.PageResult;
+import io.codelex.flightplanner.dto.SearchFlightsRequest;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@ConditionalOnProperty(prefix = "flight-planner", name = "type", havingValue = "in-DB")
+public class inDBService implements FlightServiceInterface {
+
+    private final FlightRepository flightRepository;
+    private final AirportRepository airportRepository;
+
+    public inDBService(FlightRepository flightRepository, AirportRepository airportRepository) {
+        this.flightRepository = flightRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    @Override
+    public Flight addFlight(AddFlightRequest flightRequest) {
+
+        Flight newFlight = new Flight(flightRequest.getFrom(), flightRequest.getTo(),
+                flightRequest.getCarrier(), flightRequest.getDepartureTime(), flightRequest.getArrivalTime());
+
+        airportRepository.save(newFlight.getFrom());
+        airportRepository.save(newFlight.getTo());
+
+        if (flightRepository.findAll().contains(newFlight)) {
```

Comment:
Ar findAll lietošanu DB gadījumā ir jāuzmanās - visbiežāk to nedrīkst lietot.
Ja DB būtu 10 000 ieraksti, tad viņi visu tiktu ielasīti atmiņā un tas var vienā brīdī nokārt sistēmu. Bet ir iespējams uzreiz repo meklēt pēc konkrētiem kritērijiem, vai rakstīt custom meklēšanu, kas notiek DB pusē.
Izlasi šito un arī zemāk tajā pašā rakstā ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries
Kā ari ir šāds vairants:
https://www.baeldung.com/spring-data-query-by-example#examples